torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -105,7 +105,8 @@ struct TORCH_API AccumulateGrad : public Node {
       const T& update_grad) {
     if (!variable_grad.defined()) {
       if (!GradMode::is_enabled() &&
-          !new_grad.is_sparse() &&
+          !new_grad.is_sparse() && !new_grad.is_sparse_csr() &&
+          !(variable.is_sparse_csr() && new_grad.layout() == at::kStrided) &&
```

Comment:
That third condition can never be true?

Reply:
`!(variable.is_sparse_csr() && new_grad.layout() == at::kStrided)` is needed here to avoid calling `utils::obeys_layout_contract` with sparse csr inputs.